vpn-router-api/src/main/java/vpnrouter/api/client/ClientDetectionService.java

```diff
@@ -5,6 +5,7 @@ public interface ClientDetectionService {
     void detectAndSave(CompletionListener completionListener);
 
     interface CompletionListener {
+        void onStart();
```

Comment:
No, you've broken the logic: this is a completion listener and onStart is not completion.

We may redesign the behavior.